types/lib/stdlib.go

```diff
@@ -0,0 +1,269 @@
+package lib
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	wire "github.com/cosmos/cosmos-sdk/wire"
+)
+
+// ListMapper is a Mapper interface that provides list-like functions
+// It panics when the element type cannot be (un/)marshalled by the codec
+
+type ListMapper interface {
+	// ListMapper dosen't checks index out of range
+	// The user should check Len() before doing any actions
+	Len(sdk.Context) int64
+	Get(sdk.Context, int64, interface{})
+	// Setting element out of range is harmful; use Push() when adding new elements
+	Set(sdk.Context, int64, interface{})
+	Delete(sdk.Context, int64)
+	Push(sdk.Context, interface{})
+	Iterate(sdk.Context, interface{}, func(sdk.Context, int64) bool)
+}
+
+type listMapper struct {
+	key    sdk.StoreKey
+	cdc    *wire.Codec
+	prefix string
+	lk     []byte
+}
+
+func NewListMapper(cdc *wire.Codec, key sdk.StoreKey, prefix string) ListMapper {
+	lk, err := cdc.MarshalBinary(int64(-1))
+	if err != nil {
+		panic(err)
+	}
+	return listMapper{
+		key:    key,
+		cdc:    cdc,
+		prefix: prefix,
+		lk:     lk,
+	}
+}
+
+func (lm listMapper) Len(ctx sdk.Context) int64 {
```

Comment:
Why a signed integer? Length should always be non-negative.

Reply:
To check any overflow caused by repeating `Push()`, but I'm changing it to `uint64` since it is large enough.